listenbrainz_spark/utils.py

```diff
@@ -311,3 +311,23 @@ def rename(hdfs_src_path: str, hdfs_dst_path: str):
             hdfs_dst_path – Destination path. If the path already exists and is a directory, the source will be moved into it.
     """
     hdfs_connection.client.rename(hdfs_src_path, hdfs_dst_path)
+
+
+def copy(hdfs_src_path: str, hdfs_dst_path: str, overwrite: bool = False):
+    """ Copy a file or folder in HDFS
+
+        Args:
+            hdfs_src_path – Source path.
+            hdfs_dst_path – Destination path. If the path already exists and is a directory, the source will be copied into it.
+            overwrite - Wether to overwrite the path if it already exists.
+    """
+    print(hdfs_src_path)
+    walk = hdfs_walk(hdfs_src_path)
+
+    for (root, dirs, files) in walk:
+        for _file in files:
+            src_file_path = os.path.join(root, _file)
+            dst_file_path = os.path.join(hdfs_dst_path, src_file_path.replace(hdfs_src_path, '', 1).strip("/"))
```

Comment:
what's the replace here for? Maybe this should be using the os.path functions instead?

Reply:
Changed to `os.path.relpath`.